progress.js

```diff
@@ -0,0 +1,146 @@
+"use strict";
+
+/**
+ * Helper class for dealing with HTML progress bars.
+ */
+class Progress {
+    /**
+     * `Progress :: String -> Progress`
+     *
+     * Get a reference to a progress bar by name, creating it if it does not
+     * already exist.
+     *
+     * @param {string} name Name of the progress bar.
+     */
+    constructor(name) {
+        if (Progress.progressBars[name]) {
+            return Progress.progressBars[name];
+        }
+
+        this.attached = false;
+        this.formatter = (value) => value.toString();
+        this.name = name;
+
+        Progress.progressBars[name] = this;
+    }
+
+    _updateLabel() {
+        // Cast to string before setting.
+        this._progress.attr("aria-valuenow", String(this.value));
+        this._progress.attr("aria-valuemax", String(this.max));
+        this._progressLabel.text(`${this.formattedValue()} of ${this.formattedMax()}`);
+    }
+
+    _updateView() {
+        this._progressBar.css("width", `${(this.position * 100).toFixed(2)}%`);
+    }
+
+    /**
+     * Only construct the DOM tree when it is accessed.
+     */
+    get domTree() {
+        if (!this._progress) {
+            // Only the container should be visible to ARIA. The contents should be
+            // hidden from assistive technologies.
+            const container = $("<div></div>").
```

Comment:
I'd prefer if this DOM tree construction was switched to be without jquery, something like:
```
    const container = document.createElement("div");
    container.classList.add("progress-bar");
    // attribute setting goes here...
    const bar = document.createElement("div");
    // ...
    container.append(bar);
```